### PR TITLE
[STG-2160] feat: move oclif browse CLI into monorepo (packages/cli)

### PR DESCRIPTION
**Pull request by shrey150**
## What & why

Moves the oclif-based `browse` CLI (from `browserbase/cli@0.8.2`) back into `packages/cli`, replacing the legacy commander `@browserbasehq/browse-cli`. Three things were derisked before the move (all resolved):

### 1. Independent CLI releases (no core release required)
Restored the pre-existing decoupled release machinery that PR #2143 removed when the CLI left the monorepo, repointed `@browserbasehq/browse-cli` → `browse`:
- **`release-cli.yml`** (manual dispatch): consumes CLI-only changesets, rejects mixed changesets, versions only `browse`, opens a `release/browse` PR.
- **`release.yml`**: re-added the `browse` publish / git-tag / canary steps (`npm publish --provenance`).
- **`.changeset/config.json`**: `ignore: ["browse"]` so the normal core release never touches the CLI.
- At publish, `pnpm pack` rewrites `workspace:*` → the concrete core version, so no bundling is needed.

### 2. Provable CDP-only ("local-only") build
`pnpm build:local-only` produces an artifact with **zero `BROWSERBASE_API_KEY` code paths**:
- Excludes the `cloud` / `functions` / `skills` / `templates` command + lib modules.
- All driver-side key access was centralized into `src/lib/driver/remote.ts`, which is excluded and swapped for a no-key `remote.disabled.ts` stub via a dynamic-import binding. `--remote`/auto-detect then hard-fail with a clear message.
- Verified: `grep -r BROWSERBASE_API_KEY dist/` returns nothing.

### 3. Open-source cleanup
Repointed stale `browserbase/cli` URLs (README, SKILL.md) to the monorepo path; added `license`/`author`; dropped CLI-local CI/husky/changeset/lockfile. The only hardcoded key is the public PostHog ingest key (write-only; dis/ableable via `BROWSERBASE_TELEMETRY_DISABLED`).

## Integration notes
- `@browserbasehq/stagehand: workspace:*` — CLI builds against local core; verified type-compatible with core **3.4.0** (CLI only uses `new Stagehand()`, `.init/.close/.context` + standard options).
- `turbo.json` `browse#build` task + `ci.yml --filter=browse` repointed; manifest added to build artifacts.
- Added a **scoped** `@oclif/core>minimatch: ^10.2.5` pnpm override: the monorepo's global `minimatch: ^3.1.4` pin otherwise forces oclif's minimatch to v3 and breaks `oclif manifest`. Scoped to the build-time tool only; published `browse` resolves minimatch normally.

## Known minor
- The local-only `--help` still prints the `cloud`/`functions`/etc. **topic headers** (static `package.json` metadata); the commands themselves are absent and error if invoked, and no key code ships. Left as-is to avoid per-build `package.json` mutation.

## E2E Test Matrix

| Command / flow | Observed output | Confidence / sufficiency |
| --- | --- | --- |
| `pnpm exec turbo run build --filter=browse` | core (3.4.0) + CLI build succeed; `oclif.manifest.json` written | Proves `workspace:*` against local core compiles — no API drift |
| `node bin/run.js --version` | `browse/0.8.2 darwin-arm64 node-v20.19.5` | CLI runs against workspace-linked core |
| `pnpm exec vitest run` | **200 passed (13 files)** — cloud/functions contracts, driver, doctor, telemetry, surface, oclif manifest packaging | Full surface validated post-refactor |
| `tsc --noEmit -p tsconfig.json` | clean | Type-safe against local core |
| `pnpm build:local-only` then `grep -r BROWSERBASE_API_KEY dist/` | **(no matches)** | Proves CDP-only artifact has no API-key code |
| local-only: check `dist/lib/{cloud,functions,skills,templates}` + `dist/lib/driver/remote.js` | all **absent** | Cloud modules + remote path physically excluded |
| local-only: `node bin/run.js cloud sessions list` | `Error: command cloud:sessions:list not found` | Cloud commands removed from artifact |
| local-only: `node bin/run.js open https://example.com --remote` | `Error: Remote (Browserbase) mode is disabled in this local-only build of browse.` | Remote hard-fails via stub, no key read |
| local-only: `node bin/run.js open --help` | local driver command present | Local CDP driver intact |
| `eslint` (changed driver files) | clean | — |

Closes STG-2160.

---
## Summary by cubic
Moves the oclif-based `browse` CLI into the monorepo at `packages/cli`, replacing `@browserbasehq/browse-cli`, and restores independent CLI releases. Adds a CDP-only local build (no API key), deterministic command resolution via `oclif.manifest.json`, restored CLI build checks in CI, and small test fixes; pins Prettier and bumps `hono`/`vitest` (STG-2160).

- **Refactors**
  - Restored CLI-only releases with `release-cli.yml`; re-enabled `browse` publish/tag/canary in `release.yml`; `.changeset/config.json` ignores `browse`.
  - Added `pnpm build:local-only` excluding cloud/functions/skills/templates and the remote driver; remote commands fail clearly; no `BROWSERBASE_API_KEY` code.
  - Rewired CI/Turbo to `--filter=browse`, restored CLI build checks, and cached `packages/cli/oclif.manifest.json`; `bin/run.js` boots `@oclif/core` at runtime.
  - Renamed package to `browse` (v0.8.2) with binary at `bin/run.js`; linked to core via `workspace:*`; updated README/license; scoped override `"@oclif/core>minimatch": "^10.2.5"`; pinned Prettier; bumped `hono` and `vitest`.

- **Bug Fixes**
  - Tracked `packages/cli/src/commands/cloud/sessions/downloads/get.ts` and added a root `.gitignore` negation so the command is included in CI and the manifest.
  - Added tests for the local-only build and `remote.disabled`; fixed a Vitest helper; hardened remote driver binding to only fall back on `ERR_MODULE_NOT_FOUND`; guarded browse release tag to skip tagging if already published.
  - Fixed README demo GIF URL, corrected command help examples, log glob patterns in `.gitignore`, and the evals entrypoint wiring.

<sup>Written for commit ec9b26f4789345f9bd0e6c18d4d4184277b80780. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/browserbase/stagehand/pull/2179?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/cli/src/commands/network/on.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  runDriverCommandFromFlags,
+} from "../../lib/driver/command-cli.js";
+
+export default class NetworkOn extends BrowseCommand {
+  static override description =
+    "Enable network capture for the active browser session.";
+
+  static override examples = [
+    "browse network on",
+    "browse network path",
```
[the remark this answers is not in the document]
Comment (shrey150) [reply packages/cli/src/commands/network/on.ts]: Addressed in [af0ba4bc4](https://github.com/browserbase/stagehand/commit/af0ba4bc47f5120567574fcc9f26c4a0878230cf) — removed the stray `browse network path` example so `network on` matches the sibling two-example pattern.

File: packages/cli/.gitignore [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+dist
+*.tgz
+
+# code coverage
+coverage
+*.lcov
+
+# logs
+logs
+_.log
```
[the remark this answers is not in the document]
Comment (shrey150) [reply packages/cli/.gitignore]: Addressed in [af0ba4bc4](https://github.com/browserbase/stagehand/commit/af0ba4bc47f5120567574fcc9f26c4a0878230cf) — fixed the corrupted log globs (`_.log` → `*.log`, and the adjacent `report.[0-9]*…` glob). These were copy artifacts from the upstream repo.

File: packages/cli/src/commands/click.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  driverCommandFlags,
+  runDriverCommandFromFlags,
+} from "../lib/driver/command-cli.js";
+
+export default class Click extends BrowseCommand {
+  static override description =
+    "Click an element by snapshot ref, XPath, or selector. Use `browse mouse click` for raw coordinates.";
+
+  static override examples = [
+    "browse snapshot --compact",
```
[the remark this answers is not in the document]
Comment (shrey150) [reply packages/cli/src/commands/click.ts]: Addressed in [af0ba4bc4](https://github.com/browserbase/stagehand/commit/af0ba4bc47f5120567574fcc9f26c4a0878230cf) — dropped the stray `browse snapshot --compact` example from `click`.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -90,6 +90,96 @@ jobs:
...
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Check if browse publish is needed
+        if: github.ref == 'refs/heads/main'
+        id: browse_cli
+        run: |
+          # Restore working tree — changesets/action may have switched branches
+          git checkout ${{ github.sha }}
+
+          if git diff --quiet ${{ github.sha }}^ ${{ github.sha }} -- packages/cli/package.json; then
```
[the remark this answers is not in the document]
Comment (shrey150) [reply .github/workflows/release.yml:100]: Valid edge case, but leaving as-is intentionally:

- This is the restored, previously-shipped browse-cli release logic. The `release/browse` PR (opened by `release-cli.yml`) is **squash-merged**, so the version bump lands in a single commit and `${sha}^..${sha}` detects it correctly.
- The publish step is additionally guarded by `npm view browse@${VERSION}` — so even if detection ever missed a multi-commit push, the failure mode is a *skipped* release (re-runnable via the manual `Prepare CLI Release` dispatch), never a duplicate/bad publish.
- A `github.event.before...${sha}` rewrite would need a full-history fetch and can't be exercised by this PR's CI, so it's riskier than the edge case it guards against.

Tracking as a low-priority follow-up rather than a blind change to release CI.

File: packages/cli/tests/local-only-build.test.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      const full = join(dir, entry.name);
+      return entry.isDirectory() ? walk(full) : Promise.resolve([full]);
+    }),
+  );
+  return files.flat();
+}
+
+async function exists(path: string): Promise<boolean> {
+  try {
+    await readFile(path);
```
[the remark this answers is not in the document]
Comment (shrey150) [reply packages/cli/tests/local-only-build.test.ts]: Good catch — fixed in [cd321d389](https://github.com/browserbase/stagehand/commit/cd321d389f552e515861308872cc8d972cde109d). Switched `exists()` to `fs.access()`, which correctly reports directories as existing. The excluded-directory assertions (`commands/cloud`, `lib/cloud`, …) are now meaningful rather than passing unconditionally.

**Comment by seanmcguire12** [comment]
@cubic-dev-ai review

**Review by miguelg719 — APPROVED** [review]
